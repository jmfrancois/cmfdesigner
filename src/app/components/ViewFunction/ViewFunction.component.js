import React from 'react';
import PropTypes from 'prop-types';
import { cmfConnect } from '@talend/react-cmf';
import ViewFunctionParam from '../ViewFunctionParam';


function ViewFunction(props) {
	const fn = props.fn;
	const hasKey = (fn.key && fn.name !== fn.key);
	return (
		<div>
			{hasKey && <span>{`{ ${fn.key}: `}</span>}
			<span>function{fn.generator && '*'} {fn.name}<ViewFunctionParam params={fn.params || []} /></span>
			{hasKey && <span>{' }'}</span>}
			{'{ ... }'}
		</div>
	);
}

ViewFunction.propTypes = {
	fn: PropTypes.shape({
		key: PropTypes.string,
		name: PropTypes.string,
		params: PropTypes.array,
		path: PropTypes.string,
	}),
};
ViewFunction.defaultProps = {
	fn: { params: [] },
};
ViewFunction.displayName = 'ViewFunction';

export default cmfConnect({})(ViewFunction);
