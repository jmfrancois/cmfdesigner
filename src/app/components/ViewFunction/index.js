import ViewFunction from './ViewFunction.component';

export default ViewFunction;
