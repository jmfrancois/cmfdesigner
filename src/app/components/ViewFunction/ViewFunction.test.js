import React from 'react';
import { shallow } from 'enzyme';

import ViewFunction from './ViewFunction.component';

describe('ViewFunction', () => {
	it('should render', () => {
		const wrapper = shallow(
			<ViewFunction.WrappedComponent />
		);
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
