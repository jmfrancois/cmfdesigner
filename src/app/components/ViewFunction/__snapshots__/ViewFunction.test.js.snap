// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewFunction should render 1`] = `
<div>
  <span>
    function
     
    <ViewFunctionParam
      params={Array []}
    />
  </span>
  { ... }
</div>
`;
